utils/docker/Dockerfile.leap.15

```diff
@@ -109,11 +111,10 @@ ARG DAOS_TARGET_TYPE=release
 # Now do an update to ensure software is up to date for the deps build.  If the
 # src hasn't changed then this won't do anything, but if it has then we want to
 # ensure that latest dependencies are used.
-USER root:root
```

Comment:
Ah, it's because there is no USER daos_server:daos_server above as leap does not install anything via pip so there's no need to change the user back.  I'll add this USER command back in for symmetry.